analyzers:
  # The shadow hand has two motors per EtherCAT slave device.
  # The diagnostics for the devices and motors are seperate
  # All devices are put in one category
  # The motors are split into multiple categories based on which finger they belong to.
  shadow_hand:
    type: diagnostic_aggregator/AnalyzerGroup
    path: Shadow Hand
    analyzers:
      tactiles:
        type: diagnostic_aggregator/GenericAnalyzer
        path: Tactiles
        timeout: 5.0
        regex: '([^\s]+) Tactile .*[1-5].*'
        num_items: 5
      first_finger:
        type: diagnostic_aggregator/GenericAnalyzer
        path: First Finger
        timeout: 5.0
        regex: '([^\s]+) SRDMotor .*FFJ[0-9].*'
        num_items: 5
      right_finger:
        type: diagnostic_aggregator/GenericAnalyzer
        path: Ring Finger
        timeout: 5.0
        regex: '([^\s]+) SRDMotor .*RFJ[0-9].*'
        num_items: 5
      middle_finger:
        type: diagnostic_aggregator/GenericAnalyzer
        path: Middle Finger
        timeout: 5.0
        regex: '([^\s]+) SRDMotor .*MFJ[0-9].*'
        num_items: 2
      little_finger:
        type: diagnostic_aggregator/GenericAnalyzer
        path: Little Finger
        timeout: 5.0
        regex: '([^\s]+) SRDMotor .*LFJ[0-9].*'
        num_items: 2
      thumb:
        type: diagnostic_aggregator/GenericAnalyzer
        path: Thumb
        timeout: 5.0
        regex: '([^\s]+) SRDMotor .*THJ[0-9].*'
        num_items: 4
      EtherCAT_slaves:
        type: diagnostic_aggregator/GenericAnalyzer
        path: EtherCAT Slaves
        timeout: 5.0
        regex: ['SRBridge.*', '([^\s]+) EtherCAT Dual CAN Palm']
        num_items: 2